pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -323,7 +323,7 @@ func (cb *ClusterBuilder) buildCluster(name string, discoveryType cluster.Cluste
 				c.DnsLookupFamily = cluster.Cluster_V4_ONLY
 			}
 		}
-		c.DnsRefreshRate = cb.req.Push.Mesh.DnsRefreshRate
+		c.DnsRefreshRate = protomarshal.ShallowClone(cb.req.Push.Mesh.DnsRefreshRate)
```

Comment:
this line the whole fix? the rest is the refactor for ergos and improved ShallowCopy?

Reply:
yep